### PR TITLE
HDDS-7866. [Snapshot] Fix ConcurrentModificationException in snapdiff API.

### PR DESCRIPTION
**Pull request by sadanand48**
## What changes were proposed in this pull request?
Fix CME in RocksDBCheckpointDiffer#filterRelevantSstFiles .

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-7866

## How was this patch tested?
Tested manually , existing UTS.

### CODE REVIEW

File: hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/RocksDBCheckpointDiffer.java [hunk 1]
```diff
@@ -834,11 +837,11 @@ public void filterRelevantSstFiles(Set<String> inputFiles,
               .constructBucketKey(new String(iterator.key(), UTF_8));
           if (!RocksDiffUtils
               .isKeyWithPrefixPresent(prefix, firstKey, lastKey)) {
-            inputFiles.remove(filename);
+            fileIterator.remove();
```
Comment (prashantpogde) [inline hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/RocksDBCheckpointDiffer.java:840]: Do we have some UT to cover this ?

**Review by smengcl — APPROVED** [review]

File: hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/RocksDBCheckpointDiffer.java (hunk 1 above)
Comment (smengcl) [reply hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/RocksDBCheckpointDiffer.java:840]: IIRC @sadanand48 had a UT that found this bug in the first place.

Comment (sadanand48) [reply hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/RocksDBCheckpointDiffer.java:840]: Added a unit test.

**Comment by smengcl** [comment]
Thanks @sadanand48 for the fix and UT. Thanks @prashantpogde for the review.

**Pull request merged**